### PR TITLE
Session completion

### PR DESCRIPTION
**Pull request by aizon2456**
Should fix issue #55.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,8 +3,17 @@ class ApplicationController < ActionController::Base
...
-  before_filter :current_user, :permission_check, :check_if_logged_on
+	# addresses that can be accessed while not logged in, but should not be accessed once logged in
+	def outside_path?(param)
+		outside_addresses = [
+			"/login",
+			"/signup",
+			"/"
+		]
+		outside_addresses.any?{ |path| path[param] }
+	end
```
Comment (Rosengren) [inline app/controllers/application_controller.rb]: How come this is indented?

File: app/models/user.rb [hunk 2]
```diff
@@ -6,11 +6,36 @@ class User < ActiveRecord::Base
...
 
   validates_presence_of :slug
+  attr_accessor :remember_token
 
   def to_param
     slug
   end
+
+  def set_ip(address)
+    update_attribute(:last_ip_address, "abc")
```
Comment (Rosengren) [inline app/models/user.rb:20]: why is this set to "abc"? should it be address instead?

**Comment by Rosengren** [comment]
Looks good! Seems to be working fine on my local machine. See the comments above and then merge

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (aizon2456) [inline app/controllers/application_controller.rb]: The readability was what I was aiming to achieve, changing it back to the regular spacing indentation.

File: app/models/user.rb (hunk 2 above)
Comment (aizon2456) [inline app/models/user.rb:20]: The IP logging isn't working, so I opened a new issue for it: #57.

**Pull request merged**